collect_app/src/test/java/org/odk/collect/android/instancemanagement/LocalInstancesUseCasesTest.kt

```diff
@@ -0,0 +1,164 @@
+
+        val instanceDir = File(instancesDirPath + File.separator + "Cool form name_1990-04-24_00-00-00")
+        assertThat(instanceDir.exists(), equalTo(true))
+        assertThat(instanceDir.isDirectory, equalTo(true))
+    }
+
+    @Test
+    fun `returns instance file in instance directory`() {
+        val instancesDirPath = TempFiles.createTempDir().absolutePath
+
+        val instanceFile = LocalInstancesUseCases.createInstanceFileBasedOnFormPath(
+            "/blah/blah/Cool form name.xml",
+            instancesDirPath
+        ) { 640915200000 }!!
+
+        val instanceDir = instancesDirPath + File.separator + "Cool form name_1990-04-24_00-00-00"
+        assertThat(
+            instanceFile.absolutePath,
+            equalTo(instanceDir + File.separator + "Cool form name_1990-04-24_00-00-00.xml")
+        )
+    }
+
+    @Test
+    fun `#clone makes a proper copy of the instance dir`() {
+        val instancesRepository = InMemInstancesRepository()
+        val instancesDir = TempFiles.createTempDir()
+
+        val sourceInstance = instancesRepository.save(
+            buildInstance("1", "1", instancesDir.absolutePath).build()
+        )
+        val sourceInstanceFile = File(sourceInstance.instanceFilePath)
+        val sourceInstanceFileMd5Hash = sourceInstanceFile.getMd5Hash()
+        val sourceInstanceFilePath = sourceInstanceFile.absolutePath
+
+        val sourceInstanceDir = File(sourceInstanceFile.parent!!)
+
+        val sourceMediaFile1 = TempFiles.createTempFile(sourceInstanceDir, "foo", ".jpg").also {
+            it.writeText("foo")
+        }
+        val sourceMediaFile1Md5Hash = sourceMediaFile1.getMd5Hash()
+        val sourceMediaFile1Path = sourceMediaFile1.absolutePath
+
+        val sourceMediaFile2 = TempFiles.createTempFile(sourceInstanceDir, "bar", ".mp4").also {
+            it.writeText("bar")
+        }
+        val sourceMediaFile2Md5Hash = sourceMediaFile2.getMd5Hash()
+        val sourceMediaFile2Path = sourceMediaFile2.absolutePath
+
+        val clonedInstanceDbId = LocalInstancesUseCases.clone(
+            sourceInstanceFile,
+            instancesDir.absolutePath,
+            instancesRepository
+        ) { (100_000_000_0000L..999_999_999_9999L).random() }
+
+        val clonedInstance = instancesRepository.get(clonedInstanceDbId)!!
+
+        val clonedInstanceFile = File(clonedInstance.instanceFilePath)
+        val clonedInstanceDir = File(clonedInstanceFile.parent!!)
+
+        val sourceFiles = sourceInstanceDir.listFiles()!!
+        val clonedFiles = clonedInstanceDir.listFiles()!!
+
+        assertThat(sourceFiles.size, equalTo(3))
+        assertThat(clonedFiles.size, equalTo(3))
+
+        assertThat(sourceInstanceFile.exists(), equalTo(true))
+        assertThat(clonedInstanceFile.exists(), equalTo(true))
+        assertThat(sourceInstanceFileMd5Hash, equalTo(clonedInstanceFile.getMd5Hash()))
+        assertThat(sourceInstanceFilePath, not(clonedInstanceFile))
```

Comment:
This is comparing a `File` to a `String` as far as I can see so it will always pass. I think instead we want to compare that `sourceInstanceDir` and `clonedInstanceDir` (or their paths more accurately) are not equal and that the instance file names are not equal.

Out of interest: does it actually matter if the file names are different if they live in different directories?